programs/axelar-solana-gateway/src/instructions.rs

```diff
@@ -285,19 +290,27 @@ pub fn rotate_signers(
         new_verifier_set_merkle_root,
     })?;
 
-    let mut accounts = vec![
+    let (event_authority, _bump) =
+        Pubkey::find_program_address(&[event_cpi::EVENT_AUTHORITY_SEED], &crate::ID);
+
+    let accounts = vec![
         AccountMeta::new(gateway_root_pda, false),
         AccountMeta::new_readonly(verification_session_account, false),
         AccountMeta::new_readonly(current_verifier_set_tracker_pda, false),
         AccountMeta::new(new_verifier_set_tracker_pda, false),
         AccountMeta::new(payer, true),
         AccountMeta::new_readonly(solana_program::system_program::id(), false),
+        // either push the operator as signer
+        // or push the payer as non-signer
+        if let Some(operator) = operator {
+            AccountMeta::new(operator, true)
+        } else {
+            AccountMeta::new_readonly(payer, false)
```

Comment:
I believe that if we would set an account as `Option`al with Anchor and it's not passed, it's replaced with the `program_id`. We could do the same here just to be consistent (If I understand the goal here correctly).